src/gt4py/next/embedded/nd_array_field.py

```diff
@@ -995,15 +994,15 @@ def _builtin_op(
         offset_definition = current_offset_provider[
             axis.value
         ]  # assumes offset and local dimension have same name
-        assert isinstance(offset_definition, itir_embedded.NeighborTableOffsetProvider)
+        assert common.is_neighbor_table(offset_definition)
         new_domain = common.Domain(*[nr for nr in field.domain if nr.dim != axis])
 
         broadcast_slice = tuple(
-            slice(None) if d in [axis, offset_definition.origin_axis] else xp.newaxis
+            slice(None) if d in [axis, offset_definition.domain.dims[0]] else xp.newaxis
```

Comment:
I believe, I addressed that in the overview. The problem is that the type has nice naming for this dimension, but because of how we implemented `NeighborTable` as  `Connectivity` makes it a bit hacky to expose this cleanly for the runtime object. Since the consensus seems to be: merge as soon as possible, I would work towards that in a next PR.